x-pack/plugins/security_solution/public/detection_engine/rule_management_ui/components/rules_table/bulk_actions/use_bulk_duplicate_confirmation.ts

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { useCallback, useRef } from 'react';
+
+import { useBoolState } from '../../../../../common/hooks/use_bool_state';
+
+/**
+ * hook that controls bulk duplicate actions exceptions confirmation modal window and its content
+ */
+export const useBulkDuplicateExceptionsConfirmation = () => {
+  const [isBulkDuplicateConfirmationVisible, showModal, hideModal] = useBoolState();
+  const confirmationPromiseRef = useRef<(result: string | null) => void>();
```

Comment:
Just to understand, why use `ref` here?

Reply:
I think the original idea, was to be able to wait for modal action like confirm/cancel and only then resolve the promise.

That later we can use this code like

```
const duplicateExceptions = await showBulkDuplicateConfirmation();
        if (!duplicateExceptions) {
          return;
        }
        await executeBulkAction({
          type: BulkActionType.duplicate,
          duplicatePayload: {
            include_exceptions: duplicateExceptions === DUPLICATE_OPTIONS.WITH_EXCEPTIONS,
          },
          ...(isAllSelected ? { query: filterQuery } : { ids: selectedRuleIds }),
        });
 ```